js/app.js

```diff
@@ -0,0 +1,517 @@
+
+console.log(
+  `made out of wood: ${JSON.stringify(madeOutOfWood)}\n edibles: ${JSON.stringify(edibles)}`
+);
+
+
+// 2 for loop
+function getSmallestNumber(numbers) {
+  let smallestNumber = numbers[0]
+  for (let i = 0; i < numbers.length; ++i) {
+    if (smallestNumber > numbers[i]) {
+     smallestNumber = numbers[i];
+    }
+  }
+  return smallestNumber;
+}
+
+console.log(getSmallestNumber([2, -5, 10, 1, 4])); // -5
+getSmallestNumber([200, 25, 4, 123, 87]); // 4
+
+// 2.5 sort function
+
+function giveSmallestNumber(numbers) {
+  return numbers.sort((a, b) => a - b)[0];
+}
+
+console.log(giveSmallestNumber([2, -5, 10, 1, 4])); // -5
+console.log(giveSmallestNumber([200, 25, 4, 123, 87])); // 4
+
+// 3
+console.log(getSquaredNumbers([1, 2, 3, 4, 5])); // [1, 4, 9, 16, 25]
+console.log(getSquaredNumbers([6, 7, 8, 9, 10])); // [36, 49, 64, 81, 100]
+
+function getSquaredNumbers(numbers) {
+  let squaredArray = []
+  for (let i = 0; i < numbers.length; i++) {
+    let square = numbers[i] * numbers[i];
+    squaredArray.push(square);
+  }
+  return squaredArray;
+}
+
+// function
+// make new array
+// lets multiply the array by itself
+// push it in the array
+// return the squaredArray
+
+const numbers = [1, 2, 3];
+const squaredNumbers = getSquaredNumbers(numbers);
+
+console.log(squaredNumbers); // [1, 4, 9]
+console.log(numbers); // [1, 2, 3]
+console.log(numbers !== squaredNumbers); // true
+
+// 4
+function getReversedString(string) {
+  return string.split('').reverse().join('').replace(/\s+/g, '');
+}
+
+console.log(getReversedString('Hello!')); // '!olleH'
+console.log(getReversedString('Arrays')); // 'syarrA'
+
+// 5
+// make a ispalindrome function
+// it will delete spaces and capital cases and compare it with the getreversed
+// do the same as above
+// return true if both are the same
+
+// found the delete spaces on https://stackoverflow.com/questions/5963182/how-to-remove-spaces-from-a-string-using-javascript
+// tolowerCase https://stackoverflow.com/questions/56369008/replace-capital-letters-in-an-entire-string-with-lowercase-letters-and-hyphens
+
+
+function isPalindrome(string){
+  const normaliseString = string.replace(/\s+/g, '').toLowerCase(); // made it without the spaces and capital letters
+  return normaliseString === getReversedString(string).toLowerCase();
+}
+
+console.log(isPalindrome('Kayak')); // true
+console.log(isPalindrome('Racecar')); // true
+console.log(isPalindrome('Was it a cat I saw')) // true
+console.log(isPalindrome('Hello!')); // false
+
+// 6
+// make function
+// make an empty value where the letter will be counted
+// transform the sentence so it deletes the spaces and uppercases
+// make every letter compare with one letter to make it simple in beginning
+// if true add 1 to that letter
+// return an object with the count
+// if it works, repeat for the whole alphabet
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+function countLetters(string) {
```

Comment:
please iterate over letter in a string and put the letter as a key of an object and the initial value should be 0, if the key already exists you can just increment the value by one.